test/remote/gateways/remote_stripe_payment_intents_test.rb

```diff
@@ -48,6 +48,38 @@ def test_successful_purchase
     assert purchase.params.dig('charges', 'data')[0]['captured']
   end
 
+  def test_purchases_with_same_idempotency_key
+    options = {
+      currency: 'GBP',
+      customer: @customer,
+      idempotency_key: 'test321'
+    }
+    assert purchase1 = @gateway.purchase(@amount, @visa_payment_method, options)
+    assert_equal 'succeeded', purchase1.params['status']
+    assert purchase1.params.dig('charges', 'data')[0]['captured']
+
+    assert purchase2 = @gateway.purchase(@amount, @visa_payment_method, options)
+    assert purchase2.success?
+    assert_equal purchase1.authorization, purchase2.authorization
+    assert_equal purchase1.params['charges']['data'][0]['id'], purchase2.params['charges']['data'][0]['id']
+  end
+
+  def test_purchases_with_same_idempotency_key_different_options
+    options = {
+      currency: 'GBP',
+      customer: @customer,
+      idempotency_key: 'test321'
```

Comment:
What are your thoughts on changing this and the other `idempotency_key`s in this file to a `SecureRandom.uuid` or similar? Using the same idempotency key in our remote tests over and over could lead to this failing because of logic on Stripe's end.

Reply:
@Carrigan I like it! I'll make that change now.